### PR TITLE
Update regression_org for safety against trial metadata development

### PR DESCRIPTION
**Pull request by davidmreed**
This PR updates `regression_org` to make it safe to use when the `trial` metadata is dependent upon changes in the latest beta, without losing the ability to model customer FLS in an upgrade scenario.

This PR needs additional work to define a Permission Set to permission the extra fields supplied in the `qa` config, because these fields are deployed after `update_admin_profile` has been executed.

# Critical Changes

# Changes

# Issues Closed

# Community Ideas Delivered

# New Metadata

# Deleted Metadata

### CODE REVIEW

**Comment by davidmreed** [comment]
@deepmalamehta  I just wanted to check in. Is the work to finish out updating regression_org still on the radar of the NPSP QE or dev team?

**Comment by deepmalamehta** [comment]
> @deepmalamehta I just wanted to check-in. Is the work to finish out updating regression_org still on the radar of the NPSP QE or dev team?

@davidmreed We were able to resolve this with this PR https://github.com/SalesforceFoundation/NPSP/pull/6140. It was completed in the last sprint. Thank you for checking !!

File: cumulusci.yml [hunk 1]
```diff
@@ -1013,16 +1014,22 @@ flows:
...
-                    managed: true
+                flow: config_trial
+            5:
+                task: test_data_relationships
+            6:
+                task: deploy_reports
+            7:
+                task: deploy_qa_config
+            8:
+                task: deploy_regression_bdi_metadata_config
```
Comment (davidmreed) [inline cumulusci.yml:1032]:
```suggestion
                task: deploy_regression_bdi_metadata_config
        11:
                task: assign_permission_sets
                options:
                    api_names: QA,Gift_Entry
```

**Comment by davidmreed** [comment]
Excellent. In that case I'll go ahead and mark this PR as Ready for Review, if you or one of your colleagues would like to take the revised `regression_org` for a spin and see how it goes.

**Comment by davidmreed** [comment]
@bdvorachek @skristem  This is the PR we discussed.

**Review by codexmarcel — APPROVED** [review]
CODEOWNER review passed. NPSP QE is up next!

**Review by deepmalamehta — APPROVED** [review]

**Pull request merged**